### ISSUE TITLE
Dynamic rebalancing of partition count and follower replicas

### ISSUE THREAD

**Issue by strawhat5** [zeebe-io/zeebe#4405]
**Is your feature request related to a problem? Please describe.**
Zeebe currently supports static configuration of partitions and replications, which cannot be modified dynamically for a running cluster. As of now, we need to shut down the whole cluster and clean-up the data volume in order to modify these configurations and start afresh.

**Describe the solution you'd like**
I would like to be able to dynamically modify the number of partitions and follower replicas, so that the cluster can be scaled up or down without shutting down the cluster.

**Describe alternatives you've considered**
Zeebe organises all data into [partitions](https://docs.zeebe.io/basics/partitions.html) similar to how Kafka does. Kafka has a [Rebalancing Protocol](https://www.confluent.io/blog/kafka-rebalance-protocol-static-membership/) to rebalance the clients/members to handle membership changes. Although we all know how devastating Kafka's ***stop-the-world*** rebalancing can be. So the solution that is proposed should not introduce [unnecessary rebalancing](https://www.confluent.io/blog/kafka-rebalance-protocol-static-membership/#unnecessary-rebalance) which may be detrimental to the cluster performance.

**Additional context**
[Slack archive](https://zeebe-io.slack.com/archives/C6WGNHV2A/p1588076973265100)
Depends on #4391

**Comment by npepinpe** [comment]
This is actively being worked on! 4 years later, happy to say :smile: We already have dynamic scaling of brokers (including replication factor changes), and we will add dynamic scaling of partitions in the next 2 releases. Closing in favor of existing issues.
